File: game-server/src/main/typescript/lib/service/PlayerService.ts
import { Flux, Mono } from 'reactor-core-js/flux';
import { Player } from '@shared/player_pb';
import { Location } from '@shared/location_pb';

export interface PlayerService {

    createPlayer(uuid: string, nickname: string): Player;

    disconnectPlayer(uuid: string): void;

    locate(uuid: string, locationStream: Flux<Location>): Mono<void>;

    players(): Flux<Player> ;

}

File: game-server/src/main/java/org/coinen/reactive/pacman/service/PlayerService.java
package org.coinen.reactive.pacman.service;

import org.coinen.pacman.Location;
import org.coinen.pacman.Player;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PlayerService {

    Mono<Player> createRandomPlayer(String nickname);

    Mono<Player> createGhostPlayer(String nickname);

    Mono<Player> createPacManPlayer(String nickname);

    Mono<Void> disconnectPlayer();

    Mono<Void> locate(Flux<Location> locationStream);

    Flux<Player> players();
}
